torch/nn/functional.py

```diff
@@ -1396,51 +1449,72 @@ def poisson_nll_loss(input, target, log_input=True, full=False, size_average=Tru
         full: whether to compute full loss, i. e. to add the Stirling
             approximation term. Default: ``False``
             :math:`\text{target} * \log(\text{target}) - \text{target} + 0.5 * \log(2 * \pi * \text{target})`.
-        size_average: By default, the losses are averaged over observations for
-            each minibatch. However, if the field :attr:`size_average` is set to ``False``,
-            the losses are instead summed for each minibatch. Default: ``True``
+        size_average (bool, optional): Deprecated (see :attr:`reduction`). By default,
```

Comment:
here and other docs: do you want to deprecate `size_average` and `reduce`? If so, I have two suggestions:
1. move them below `reduction` in doc
2. raise deprecation warning if people use old api.

Reply:
OK to 1.

For number 2. I do raise a deprecation warning. It's in get_reduction_constant in the Reduction class. I call it whenever whenever the nn.functional function or nn module is passed size_average or reduce.

Reply:
I'm actually doing them in this order because I want them to match the actual order of the args.